pkg/cmd/roachtest/tests/large_schema_backup_restore.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/spec"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/stretchr/testify/require"
+)
+
+// registerLargeSchemaBackupRestores registers tests that large schemas can
+// backup/restore successfully.
+func registerLargeSchemaBackupRestores(r registry.Registry) {
+	// 10k is probably the upper limit right now, but we haven't tested further.
+	largeSchemaBackupRestore(r, 10000)
+}
+
+func largeSchemaBackupRestore(r registry.Registry, numTables int) {
```

Comment:
nit: provide a one or 2 sentence description of the workload this test runs?

Reply:
👍